talented-tigers/spydir/generate_page.py

```diff
@@ -0,0 +1,200 @@
+import random
+import re
+import wikipedia
+import nltk
+import threading
+from faker import Faker
+from .generate_images import generate_images
+import gpt_2_simple as gpt2
+from .models import GeneratedPage, BlogPost
+
+
+def generate_page(page_name, page_type=None):
+    """Gets a page object which only has a title, then populates it with scraped information"""
+    page_object = GeneratedPage.objects.get(page_title=page_name)
+
+    possible_page_types = [page_type[0] for page_type in GeneratedPage.page_type_choices]
+    # Chooses a random page type from a list of all page types. weights are in this order:
+    # BLOG, INFO, BIZ
+    page_object.page_type = (
+        random.choices(possible_page_types, [0.25, 0.5, 0.25])[0] if page_type is None else page_type
+    )
+
+    # Define the different fields needed for different page types here
+    if page_object.page_type == "BLOG":
+        page_object.page_author = generate_page_author()
+        page_object.blogger_age = random.randrange(8, 95)
+        page_object.blogger_location = generate_blogger_location()
+
+        # Generates x amount of blog posts
+        no_posts = (int(str(page_object.css_seed)[0]) // 2) + 1
+        titles = []
+        for x in range(no_posts):
+            title = generate_blog_post_title(page_name)
+            titles.append(title)
+            blog_post = BlogPost.objects.create(title=title, content="Loading...")
+            blog_post.save()
+            page_object.blog_posts.add(blog_post)
+
+        post_thread = threading.Thread(target=generate_gpt2, args=(page_object.blog_posts,))
+        post_thread.start()
+
+    elif page_object.page_type == "INFO":
+        wikipedia_page = generate_information(page_name)
+        page_object.page_content = wikipedia_page[0]
+        page_object.page_source_url = wikipedia_page[1]
+        page_object.page_author = generate_page_author()
+
+        page_images = generate_images(False, 1, page_name)
+        for page_image in page_images:
+            page_image.save()
+            page_object.page_images.add(page_image)
+
+    elif page_object.page_type == "BIZ":
+        faker = Faker()
+        page_object.business_phone_num = faker.phone_number()
+        page_object.business_email = faker.email()
+
+        page_images = generate_images(False, 1, page_name)
```

Comment:
Magic numbers are often a bit unclear, maybe keyword arguments would have made it more clear to me.